### PR TITLE
Enabling `TF` on `image-classification` pipeline.

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@philschmid 
@LysandreJik

### CODE REVIEW

File: src/transformers/pipelines/image_classification.py [hunk 1]
```diff
@@ -87,8 +98,16 @@ def _forward(self, model_inputs):
     def postprocess(self, model_outputs, top_k=5):
         if top_k > self.model.config.num_labels:
             top_k = self.model.config.num_labels
-        probs = model_outputs.logits.softmax(-1)[0]
-        scores, ids = probs.topk(top_k)
+
+        if self.framework == "pt":
```
Comment (philschmid) [inline src/transformers/pipelines/image_classification.py:102]: Would it make more sense to make the post-processing framework-agnostic? similar to the `text-classification` pipeline? https://github.com/huggingface/transformers/blob/f2ab21833f3c1fcc8dea76988bdacd368fca779e/src/transformers/pipelines/text_classification.py#L20

Comment (Narsil) [reply src/transformers/pipelines/image_classification.py:102]: There's no `top_k` with a nice API on `numpy` so I would stay away from `native`-> `np` -> `native` or have the ugly `topk` version in `np`.

Overall this looks to me as the least worst of all codes.

Do you know a clean `np` version of `topk` (would indeed make things cleaner)

Comment (LysandreJik) [reply src/transformers/pipelines/image_classification.py:102]: I think one can do a top_k using `np.argpartition` and/or `np.argsort`, do you mean that there's no `np.topk` when you say there's no clean version?

Comment (Narsil) [reply src/transformers/pipelines/image_classification.py:102]: I mean  `np.argsort/np.argpartition` is not as clean as `torch.top_k`.

`top_k` can be done in `O(k log(n))` where sorting is `O(n + k log(k))` with numpy apparently:

https://stackoverflow.com/questions/6910641/how-do-i-get-indices-of-n-maximum-values-in-a-numpy-array/23734295#23734295

```python
ind = np.argpartition(a, -top_k)[-top_k:]
indices = ind[np.argsort(a[ind])]
values = a[indices]
```
is also pretty unreadable. 

I don't think having a single `if framework == "tf"` in the decoding is that bad.

**Review by LysandreJik — APPROVED** [review]
Ok then LGTM

Thanks for working on it @Narsil

File: src/transformers/pipelines/image_classification.py (hunk 1 above)
Comment (philschmid) [reply src/transformers/pipelines/image_classification.py:102]: That's true, but if we want to enable pipelines for `jax` or re-use parts of it in `optimum` i think we should try to have an aligned way of doing things and not use for one pipeline framework-specific (pt, tf)processing and for another pipeline not

**Pull request merged**